src/addon/validation.cpp

```diff
@@ -44,6 +45,27 @@ namespace {
 			}
 		}
 	};
+
+	struct addon_filename_ucs4char_illegal
+	{
+		inline bool operator()(ucs4::char_t c) const
+		{
+			switch(c){
+			case ' ':
+			case '/':
+			case ':':
+			case '\\':
+			case '~':
+			case 0x7F: // DEL
+				return true;
+			default:
+				return (
+					c < 0x20 || // control characters
+					(c >= 0xD800 && c < 0xE000) // surrogate pairs
```

Comment:
Nitpicking: I'd prefer if the comment said just "surrogates" instead of "surrogate pairs", as it analyses only one code point at a time.